server/src/config.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) ZeroC, Inc.
+
+use crate::Backend;
+use tower_lsp::lsp_types::{ConfigurationItem, DidChangeConfigurationParams, Url};
+
+#[derive(Default, Debug)]
+pub struct SliceConfig {
+    pub reference_urls: Option<Vec<Url>>,
+}
+
+impl SliceConfig {
+    // Attempt to create a SliceConfig from the backend with the provided uri root.
+    pub async fn try_from_backend(
+        backend: &Backend,
```

Comment:
I think all we want here is the client. Or really just the configuration already fetched from the client.